crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -676,40 +678,64 @@ where
                 value,
                 ..
             }) => {
-                // If we're in a class or module scope, then the annotation needs to be
-                // available at runtime.
-                // See: https://docs.python.org/3/reference/simple_stmts.html#annotated-assignment-statements
-                let runtime_annotation = if self.semantic.future_annotations() {
-                    self.semantic
+                enum AnnotationKind {
+                    RuntimeRequired,
+                    RuntimeEvaluated,
+                    TypingOnly,
+                }
+
+                fn annotation_kind(
+                    semantic: &SemanticModel,
+                    settings: &LinterSettings,
+                ) -> AnnotationKind {
+                    // If the annotation is in a class, and that class is marked as
+                    // runtime-evaluated, treat the annotation as runtime-required.
+                    // TODO(charlie): We could also include function calls here.
+                    if semantic
                         .current_scope()
                         .kind
                         .as_class()
                         .is_some_and(|class_def| {
-                            flake8_type_checking::helpers::runtime_evaluated_class(
+                            flake8_type_checking::helpers::runtime_required_class(
```

Comment:
It's unfortunate that we now have runtime **required** internally but runtime **evaluated** in the user facing option. Maybe we could use either `v0.2.0` or `v0.3.0` as a way to rename these options.

Reply:
Yes great observation! It's really unfortunate.

Reply:
Actually, I'm gonna change the internal names.